gems/smithy/lib/smithy/anvil/templates/client/plugins/endpoint.erb

```diff
@@ -0,0 +1,33 @@
+# frozen_string_literal: true
+
+# This is generated code!
+
+module <%= namespace %>
+  module Plugins
+    class Endpoint < Smithy::Client::Plugin
+      option(
+        :endpoint_provider,
+        doc_type: '<%= namespace %>::EndpointProvider',
+        docstring: <<~DOCS) do |_cfg|
+The endpoint provider used to resolve endpoints. Any object that responds to
+`#resolve_endpoint(parameters)` where `parameters` is a Struct similar to
+`Aws::ACM::EndpointParameters`.
```

Comment:
This plugin lives in generic side of smithy - should we have this specific reference to an AWS service?

Reply:
Yeah good call, I removed that.  I